ircd/send.c

```diff
@@ -1163,34 +1163,33 @@ sendto_local_clients_with_capability(int cap, const char *pattern, ...)
 	msgbuf_cache_free(&msgbuf_cache);
 }
 
-/* sendto_monitor()
+/*
+ * _sendto_monitor_with_capability_butserial()
  *
- * inputs	- monitor nick to send to, format, va_args
- * outputs	- message to local users monitoring the given nick
- * side effects -
+ * Shared implementation of sendto_monitor_with_capability_butserial and _sendto_monitor
  */
 void
-sendto_monitor(struct Client *source_p, struct monitor *monptr, const char *pattern, ...)
+_sendto_monitor_with_capability_butserial(struct Client *source_p, struct monitor *monptr, int caps, int negcaps, bool skipserial, const char *pattern, va_list * args)
 {
-	va_list args;
 	struct Client *target_p;
 	rb_dlink_node *ptr;
 	rb_dlink_node *next_ptr;
 	struct MsgBuf msgbuf;
 	struct MsgBuf_cache msgbuf_cache;
-	rb_strf_t strings = { .format = pattern, .format_args = &args, .next = NULL };
+	rb_strf_t strings = { .format = pattern, .format_args = args, .next = NULL };
 
 	build_msgbuf_tags(&msgbuf, source_p);
 
-	va_start(args, pattern);
 	msgbuf_cache_init(&msgbuf_cache, &msgbuf, &strings);
-	va_end(args);
 
 	RB_DLINK_FOREACH_SAFE(ptr, next_ptr, monptr->users.head)
 	{
 		target_p = ptr->data;
 
-		if(IsIOError(target_p))
+		if(IsIOError(target_p) ||
+			(skipserial && target_p->serial == current_serial) ||
+			!IsCapable(target_p, caps) ||
```

Comment:
Eliminate current_serial abuse

```suggestion
		if(IsIOError(target_p) ||
			!IsCapable(target_p, caps) ||
```